Description: Rename binary file from docker to docker.io
Author: Tianon Gravi <admwiggin@gmail.com>
Forwarded: not-needed

diff --git a/contrib/completion/bash/docker b/contrib/completion/bash/docker
index 1449330..b658896 100755
--- a/contrib/completion/bash/docker
+++ b/contrib/completion/bash/docker
@@ -22,7 +22,7 @@
 # must have access to the socket for the completions to function correctly
 
 __docker_q() {
-	docker 2>/dev/null "$@"
+	docker.io 2>/dev/null "$@"
 }
 
 __docker_containers_all()
@@ -689,4 +689,4 @@ _docker()
 	return 0
 }
 
-complete -F _docker docker
+complete -F _docker docker.io
