Description: Rename binary file from docker to docker.io
Author: Prach Pongpanich <prachpub@gmail.com>
Forwarded: not-needed

--- a/contrib/completion/bash/docker
+++ b/contrib/completion/bash/docker
@@ -24,38 +24,38 @@
 __docker_containers_all()
 {
 	local containers
-	containers="$( docker ps -a -q )"
-	names="$( docker inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
+	containers="$( docker.io ps -a -q )"
+	names="$( docker.io inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
 	COMPREPLY=( $( compgen -W "$names $containers" -- "$cur" ) )
 }
 
 __docker_containers_running()
 {
 	local containers
-	containers="$( docker ps -q )"
-	names="$( docker inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
+	containers="$( docker.io ps -q )"
+	names="$( docker.io inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
 	COMPREPLY=( $( compgen -W "$names $containers" -- "$cur" ) )
 }
 
 __docker_containers_stopped()
 {
 	local containers
-	containers="$( comm -13 <(docker ps -q | sort -u) <(docker ps -a -q | sort -u) )"
-	names="$( docker inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
+	containers="$( comm -13 <(docker.io ps -q | sort -u) <(docker.io ps -a -q | sort -u) )"
+	names="$( docker.io inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
 	COMPREPLY=( $( compgen -W "$names $containers" -- "$cur" ) )
 }
 
 __docker_image_repos()
 {
 	local repos
-	repos="$( docker images | awk 'NR>1{print $1}' )"
+	repos="$( docker.io images | awk 'NR>1{print $1}' )"
 	COMPREPLY=( $( compgen -W "$repos" -- "$cur" ) )
 }
 
 __docker_images()
 {
 	local images
-	images="$( docker images | awk 'NR>1{print $1":"$2}' )"
+	images="$( docker.io images | awk 'NR>1{print $1":"$2}' )"
 	COMPREPLY=( $( compgen -W "$images" -- "$cur" ) )
 	__ltrim_colon_completions "$cur"
 }
@@ -63,8 +63,8 @@
 __docker_image_repos_and_tags()
 {
 	local repos images
-	repos="$( docker images | awk 'NR>1{print $1}' )"
-	images="$( docker images | awk 'NR>1{print $1":"$2}' )"
+	repos="$( docker.io images | awk 'NR>1{print $1}' )"
+	images="$( docker.io images | awk 'NR>1{print $1":"$2}' )"
 	COMPREPLY=( $( compgen -W "$repos $images" -- "$cur" ) )
 	__ltrim_colon_completions "$cur"
 }
@@ -72,9 +72,9 @@
 __docker_containers_and_images()
 {
 	local containers images
-	containers="$( docker ps -a -q )"
-	names="$( docker inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
-	images="$( docker images | awk 'NR>1{print $1":"$2}' )"
+	containers="$( docker.io ps -a -q )"
+	names="$( docker.io inspect -format '{{.Name}}' $containers | sed 's,^/,,' )"
+	images="$( docker.io images | awk 'NR>1{print $1":"$2}' )"
 	COMPREPLY=( $( compgen -W "$images $names $containers" -- "$cur" ) )
 	__ltrim_colon_completions "$cur"
 }
@@ -544,4 +544,4 @@
 	return 0
 }
 
-complete -F _docker docker
+complete -F _docker docker.io
