Description: Rename binary file from docker to docker.io
Author: Tianon Gravi <admwiggin@gmail.com>
Forwarded: not-needed

--- a/contrib/init/systemd/docker.service
+++ b/contrib/init/systemd/docker.service
@@ -7,7 +7,7 @@ After=multi-user.target
 [Service]
 Type=simple
 ExecStartPre=/bin/mount --make-rprivate /
-ExecStart=/usr/bin/docker -d
+ExecStart=/usr/bin/docker.io -d
 
 [Install]
 WantedBy=multi-user.target
