Description: Sysvinit needs to "Provides" docker.io, not docker
Author: Tianon Gravi <admwiggin@gmail.com>
Forwarded: not-needed

--- a/contrib/init/sysvinit/docker
+++ b/contrib/init/sysvinit/docker
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 ### BEGIN INIT INFO
-# Provides:           docker
+# Provides:           docker.io
 # Required-Start:     $syslog $remote_fs
 # Required-Stop:      $syslog $remote_fs
 # Default-Start:      2 3 4 5
